#! /bin/sh
patch -p1 -f $* < $0
exit $?

commit b38ded6edc058004adfbbc88149c4aea6c23aad7
Author: Carl Schwan <carl@carlschwan.eu>
Date:   Wed Jun 11 11:52:30 2025 +0200

    Connect only once to attachment list double click event

    Otherwise the attachment get opened multiple times when double clicking
    on it after navigating to a different email.

diff --git a/src/widgets/messageviewer.cpp b/src/widgets/messageviewer.cpp
index 4bb2d7a..6c267b1 100644
--- a/src/widgets/messageviewer.cpp
+++ b/src/widgets/messageviewer.cpp
@@ -192,6 +192,28 @@ MessageViewer::MessageViewer(QWidget *parent)
         d->selectionChanged();
         d->showContextMenu();
     });
+
+    connect(d->parser.attachments(), &AttachmentModel::info, this, [this](const QString &message) {
+        d->messageWidget->setMessageType(KMessageWidget::Information);
+        d->messageWidget->setText(message);
+        d->messageWidget->animatedShow();
+    });
+
+    connect(d->parser.attachments(), &AttachmentModel::errorOccurred, this, [this](const QString &message) {
+        d->messageWidget->setMessageType(KMessageWidget::Error);
+        d->messageWidget->setText(message);
+        d->messageWidget->animatedShow();
+    });
+
+    connect(d->attachmentView->selectionModel(), &QItemSelectionModel::selectionChanged, this, [this] {
+        d->selectionChanged();
+    });
+
+    connect(d->attachmentView, &QAbstractItemView::doubleClicked, this, [this](const QModelIndex &) {
+        // Since this is only emitted if a valid index is double clicked we can assume
+        // that the first click of the double click set the selection accordingly.
+        d->openSelectedAttachments();
+    });
 }

 MessageViewer::~MessageViewer()
@@ -369,18 +391,6 @@ void MessageViewer::setMessage(const KMime::Message::Ptr message)
     setUpdatesEnabled(false);
     d->parser.setMessage(message);

-    connect(d->parser.attachments(), &AttachmentModel::info, this, [this](const QString &message) {
-        d->messageWidget->setMessageType(KMessageWidget::Information);
-        d->messageWidget->setText(message);
-        d->messageWidget->animatedShow();
-    });
-
-    connect(d->parser.attachments(), &AttachmentModel::errorOccurred, this, [this](const QString &message) {
-        d->messageWidget->setMessageType(KMessageWidget::Error);
-        d->messageWidget->setText(message);
-        d->messageWidget->animatedShow();
-    });
-
     for (int i = d->formLayout->rowCount() - 1; i >= 0; i--) {
         d->formLayout->removeRow(i);
     }
@@ -422,16 +432,6 @@ void MessageViewer::setMessage(const KMime::Message::Ptr message)
     d->attachmentView->setModel(d->parser.attachments());
     d->attachmentView->setVisible(d->parser.attachments()->rowCount() > 0);

-    connect(d->attachmentView->selectionModel(), &QItemSelectionModel::selectionChanged, this, [this] {
-        d->selectionChanged();
-    });
-
-    connect(d->attachmentView, &QAbstractItemView::doubleClicked, this, [this](const QModelIndex &) {
-        // Since this is only emitted if a valid index is double clicked we can assume
-        // that the first click of the double click set the selection accordingly.
-        d->openSelectedAttachments();
-    });
-
     setUpdatesEnabled(true);
 }
